crates/autopilot/src/run_loop.rs

```diff
@@ -121,9 +124,68 @@ impl RunLoop {
         driver: &Driver,
         _solution: &solve::Response,
     ) -> Result<()> {
-        let request = execute::Request { auction_id: id };
+        let request = execute::Request {
+            auction_id: id,
+            tag: id.to_be_bytes().into(),
+        };
         let _response = driver.execute(&request).await.context("execute")?;
-        // TODO: Wait for transaction to be mined or deadline to be reached.
+        // TODO: React to deadline expiring.
+        self.wait_for_settlement_transaction(&request.tag)
+            .await
+            .context("wait for settlement transaction")?;
         Ok(())
     }
+
+    /// Tries to find a `settle` contract call with calldata ending in `tag`.
+    ///
+    /// Returns None if no transaction was found within the deadline.
+    pub async fn wait_for_settlement_transaction(&self, tag: &[u8]) -> Result<Option<Transaction>> {
+        // Start earlier than current block because there might be a delay when receiving the
+        // Solver's /execute response during which it already started broadcasting the tx.
+        let start_offset = 10;
+        let max_wait_time = 20;
+        let current = self.current_block.borrow().number;
+        let start = current.saturating_sub(start_offset);
+        let deadline = current.saturating_add(max_wait_time);
```

Comment:
This needs to be configurable because different networks have different block throughput. Will do in future PR.

Reply:
I think that the `tokio::sleep` also needs to be configurable for the same reason (different block rates on different chains).